config.toml.example

```diff
@@ -185,47 +182,56 @@ changelog-seen = 2
 # The default stage to use for the `bench` subcommand
 #bench-stage = 2
 
-# Build triple for the original snapshot compiler. This must be a compiler that
-# nightlies are already produced for. The current platform must be able to run
-# binaries of this build triple and the nightly will be used to bootstrap the
-# first compiler.
+# Build triple for the pre-compiled snapshot compiler. If `rustc` is set, this must match its host
+# triple (see `rustc --version --verbose`; cross-compiling the rust build system itself is NOT
+# supported). If `rustc` is unset, this must be a platform with pre-compiled host tools
+# (https://doc.rust-lang.org/nightly/rustc/platform-support.html). The current platform must be
+# able to run binaries of this build triple.
 #
-# Defaults to platform where `x.py` is run.
+# If `rustc` is present in path, this defaults to the host it was compiled for.
+# Otherwise, `x.py` will try to infer it from the output of `uname`.
```

Comment:
What happens on Windows?

Reply:
We hard-code MSVC: https://github.com/rust-lang/rust/blob/1057e2132e32764676e89e88992dc7a4c3e955ea/src/bootstrap/bootstrap.py#L226-L232

Reply:
Does msys2/Whatever-is-the-main-environnement-used-to-build-the-`*-pc-windows-gnu`-toolchains have `uname` working?

Reply:
I don't know. I would rather not block this PR on fixing all of bootstrap's bugs.

Reply:
I read the code a bit below, and it seems like it can detect the presence of `uname` and use it, but I think adding a note that when `uname` does not exist/does not return an expected value on Windows, it defaults to MSVC.